### PR TITLE
MODDICONV-209 Create default Action profile and Mapping profile

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
## Purpose
Create default Action profile and Mapping profile for Authority

## Approach
 - Create default Action profile, "folioRecord": "AUTHORITIES", link it to the JobProfile profile ("id": "6eefa4c6-bbf7-4845-ad82-de7fc5abd0e3",)
 - Create default Mapping profile, "existingRecordType": "AUTHORITIES", "incomingRecordType": "MARC_AUTHORITIES", link it to the Action profile
 - Update FolioRecord enum, add "AUTHORITIES" there
 - Add validation to the REST API (PUT and DELETE methods)

## Learning
[MODDICONV-209](https://issues.folio.org/browse/MODDICONV-209)

### CODE REVIEW

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 1]
```diff
@@ -22,3 +22,82 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
 }') ON CONFLICT DO NOTHING;
+
+INSERT INTO ${myuniversity}_${mymodule}.action_profiles (id, jsonb) values
+  ('7915c72e-c6af-4962-969d-403c7238b051', '{
+  "id": "7915c72e-c6af-4962-969d-403c7238b051",
+  "name": "Default - Create Authorities",
+  "action": "CREATE",
+  "deleted": false,
+  "description": "This action profile is used with FOLIO''s default job profile for creating Inventory Authorities and SRS MARC Authorities records. It can be edited, duplicated.",
+  "folioRecord": "AUTHORITIES",
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:33]: AUTHORITY

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 2]
```diff
@@ -22,3 +22,82 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+  "userInfo": {
+    "lastName": "System",
+    "userName": "System",
+    "firstName": "System"
+  },
+  "description": "This field mapping profile is used with FOLIO''s default job profile for creating Inventory Authorities and SRS MARC Authorities records. It can be edited, duplicated, deleted, or linked to additional action profiles.",
+  "childProfiles": [],
+  "mappingDetails": {
+    "name": "authorities",
+    "recordType": "AUTHORITIES",
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:70]: AUTHORITY

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 3]
```diff
@@ -22,3 +22,82 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+    "mappingFields": [],
+    "marcMappingDetails": []
+  },
+  "parentProfiles": [],
+  "existingRecordType": "AUTHORITIES",
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:75]: AUTHORITY

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 4]
```diff
@@ -22,3 +22,82 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+  "incomingRecordType": "MARC_AUTHORITIES",
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:76]: MARC_AUTHORITY

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 5]
```diff
@@ -22,3 +22,82 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+    "lastName": "System",
+    "userName": "System",
+    "firstName": "System"
+  },
+  "description": "This field mapping profile is used with FOLIO''s default job profile for creating Inventory Authorities and SRS MARC Authorities records. It can be edited, duplicated, deleted, or linked to additional action profiles.",
+  "childProfiles": [],
+  "mappingDetails": {
+    "name": "authorities",
+    "recordType": "AUTHORITIES",
+    "mappingFields": [],
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:71]: No mapping for now?

**Review by psmagin — CHANGES_REQUESTED** [review]

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 6]
```diff
@@ -22,3 +22,144 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+        "subfields": []
+      }, {
+        "name": "genreTerm",
+        "path": "authority.genreTerm",
+        "value": "",
+        "enabled": "false",
+        "subfields": []
+      }, {
+        "name": "identifiers",
+        "path": "authority.identifiers",
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:122]:
```suggestion
        "path": "authority.identifiers[]",
```
I thinks it's also an array like notes

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 7]
```diff
@@ -22,3 +22,144 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
     "updatedByUserId": "00000000-0000-0000-0000-000000000000"
   }
 }') ON CONFLICT DO NOTHING;
+
+INSERT INTO ${myuniversity}_${mymodule}.action_profiles (id, jsonb) values
```
Comment (oleksandr-dekin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:26]: Can we delete action action profile? Or we should restrict this possibility, the same as for derive marc records.

File: mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql [hunk 8]
```diff
@@ -22,3 +22,144 @@ INSERT INTO ${myuniversity}_${mymodule}.job_profiles (id, jsonb) values
...
+    },
+    "userInfo": {
+      "lastName": "System",
+      "userName": "System",
+      "firstName": "System"
+    }
+}')
+ON CONFLICT DO NOTHING;
+
+INSERT INTO ${myuniversity}_${mymodule}.mapping_profiles (id, jsonb) values
```
Comment (oleksandr-dekin) [inline mod-data-import-converter-storage-server/src/main/resources/templates/db_scripts/defaultData/default_marc_authority_job_profile.sql:50]: Same question as above

**Pull request merged**